### ISSUE TITLE
Error with the `list()` construct

### ISSUE THREAD

**Issue by mirko-pagliai** [cakephp/cakephp-codesniffer#158]
I've this code:

    list(,, $extension, $filename) = array_values(pathinfo($filename));

And this related error:

    57 | ERROR | [x] Missing space after comma

Also trying this (spaces before commas):

    list( , , $extension, $filename) = array_values(pathinfo($filename));

I've this error:

    57 | ERROR | Space before comma, expected none, though
    57 | ERROR | Space before comma, expected none, though

Where am I doing wrong?

**Comment by dereuromark** [comment]
Might be a bug in the sniffer rule.
Check my https://github.com/php-fig-rectified/psr2r-sniffer/blob/master/PSR2R/Sniffs/PHP/ListCommaSniff.php if that one would fix the issue, you can port the fix over as a PR if you want to in case that resolves it.

**Comment by ADmad** [comment]
@mirko-pagliai Run phpcs with `-s` switch so that it shows the name of the sniff which is reporting this error.

**Comment by mirko-pagliai** [comment]
@ADmad 

    57 | ERROR | [x] Missing space after comma
       |       |     (CakePHP.WhiteSpace.CommaSpacing)

and again:

    57 | ERROR | Space before comma, expected none, though
       |       | (CakePHP.WhiteSpace.CommaSpacing)
    57 | ERROR | Space before comma, expected none, though
       |       | (CakePHP.WhiteSpace.CommaSpacing)

Now I'll try the fix proposed by @dereuromark .

**Comment by mirko-pagliai** [comment]
@dereuromark with no space

    list(,, $extension, $filename) = array_values(pathinfo($filename));

 I've the same error:

    57 | ERROR | [x] Missing space after comma
     |       |     (PSR2R.WhiteSpace.CommaSpacing)

Adding a space:

    list(, , $extension, $filename) = array_values(pathinfo($filename));

no errors, ok.

Now, what I have to do? I don't know how a sniffer works.

**Comment by dereuromark** [comment]
I would expect it to have a space to make it easier to count.
Someone can just port my CommaSpacing fixer changes to the Cake one.

**Comment by mirko-pagliai** [comment]
I am not able, I'm sorry. I wait, then.

**Comment by dereuromark** [comment]
It is basically just adding
```php
			if ($tokens[$previous]['code'] === T_COMMA) {
				return;
			}
```
OK, we can do it.

**Comment by dereuromark** [comment]
PR open.
